src/aosm/azext_aosm/generate_nfd/vnf_nfd_generator.py

```diff
@@ -280,18 +292,15 @@ def write_vhd_parameters(self, directory: Path) -> None:
 
         :param directory: The directory to put this file in.
         """
-        azureDeployLocation: str
-        if self.vm_parameters.get("location"):
-            # Location can be passed in as deploy parameter
-            azureDeployLocation = "{deployParameters.location}"
-        else:
-            # Couldn't find a location parameter in the source template, so hard code to
-            # the location we are deploying the publisher to.
-            azureDeployLocation = self.config.location
-
+        vhd_config = self.config.vhd
         vhd_parameters = {
             "imageName": self.image_name,
-            "azureDeployLocation": azureDeployLocation,
+            "azureDeployLocation": "{deployParameters.location}",
+            **{
```

Comment:
This looks very neat and quick, but prone to being unwittingly broken in furture. I'd prefer you did it long-hand, so if someone decides to add some mad new image config parameter which doesn't go here, it doesn't break the NFDV.

Reply:
I'm probably ok with it if you just change the "if key.startswith("image"), to specifically state the keys we are interested in.